support/cas-server-support-openrewrite/build.gradle

```diff
@@ -1,38 +1,38 @@
 description = "Apereo CAS Open Rewrite Support"
 
-processResources {
-    def casVersion = project.version
+tasks.register("createRecipe") {
+    def casVersion = rootProject.version
     def tomcatVersion = project.tomcatVersion
     def sourceCompatibility = project.sourceCompatibility
     def springBootVersion = project.springBootVersion
     def jibVersion = project.jibVersion
     def gradleVersion = gradle.gradleVersion
     def recipeName = "cas${casVersion.replaceAll(/\.|-SNAPSHOT/, '')}"
-    def templateContents = file("src/main/resources/current.gtemplate").text
 
-    def outputDir = file("src/main/resources/META-INF/rewrite")
-    def outputFile = file("${outputDir.absolutePath}/${recipeName}.yml")
+    def inputFile = file("src/main/resources/current.gtemplate")
+    def templateContents = inputFile.text
+    def recipeFileName = "$buildDir/generated/META-INF/rewrite/${recipeName}.yml"
+    def outputFile = file(recipeFileName)
+
     outputs.file(outputFile)
+    outputs.cacheIf { true }
```

Comment:
If you want to enable caching only do so if you've added ALL of inputs (and outputs) to be tracked. This includes all the versions and anything else that might affect the generated file(s) that are marked as outputs.